File: 8_kyu/Neutralisation.py
def neutralise(s1 :str, s2 :str) -> str:
    """
    Merges two strings, returns a string of equal characters and "0" at positions of different characters.
    s1 -- first string to merge
    s2 -- second string to merge
    Return -- string with equal chars of s1 and s2 and "0" where chars not equal.
    """
    return "".join( c[0] if c[0]==c[1] else "0" for c in zip(s1,s2) )

File: 8_kyu/Neutralisation.rs
/// Neutralisation of string differences.
///
/// The function returns a new string.
/// Each character is copied from s1 and s2 if the corresponding (same position) characters are the same.
/// Otherwise the character '0' is placed in the resulting string.
/// So the differences in the strings are neutralised to '0' in the result string.
fn neutralise(s1: &str, s2: &str) -> String {
    s1.chars()
    .zip(s2.chars())
    .map(|ab| if ab.0 == ab.1 {ab.0} else {'0'} )
    .collect()
}
